firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java

```diff
@@ -119,7 +134,111 @@ public void onFailure(@NonNull @NotNull Exception e) {
    */
   @NonNull
   public Task<AppDistributionRelease> checkForUpdate() {
-    return Tasks.forResult(null);
+
+    if (checkForUpdateTaskCompletionSource != null
+        && !checkForUpdateTaskCompletionSource.getTask().isComplete()) {
+      checkForUpdateCancellationSource.cancel();
+    }
+
+    checkForUpdateCancellationSource = new CancellationTokenSource();
+    checkForUpdateTaskCompletionSource =
+        new TaskCompletionSource<>(checkForUpdateCancellationSource.getToken());
+    checkForUpdateHandler = HandlerCompat.createAsync(Looper.getMainLooper());
+
+    firebaseInstallationsApi
+        .getId()
+        .addOnSuccessListener(
+            new OnSuccessListener<String>() {
+              @Override
+              public void onSuccess(String fid) {
+                firebaseInstallationsApi
+                    .getToken(true)
+                    .addOnSuccessListener(
+                        new OnSuccessListener<InstallationTokenResult>() {
+                          @Override
+                          public void onSuccess(InstallationTokenResult installationTokenResult) {
+
+                            String authToken = installationTokenResult.getToken();
+                            String appId = firebaseApp.getOptions().getApplicationId();
+                            String apiKey = firebaseApp.getOptions().getApiKey();
+
+                            // make into method parameters
+                            FirebaseAppDistributionTesterApiClient client =
+                                new FirebaseAppDistributionTesterApiClient();
+
+                            handleLatestReleaseFromClient(client, fid, appId, apiKey, authToken);
+                          }
+                        });
+              }
+            })
+        .addOnFailureListener(
+            new OnFailureListener() {
+              @Override
+              public void onFailure(@NonNull Exception e) {
+                setCheckForUpdateTaskCompletionError(
+                    new FirebaseAppDistributionException(
```

Comment:
I think it might be better to return a signin failure error than an unknown error since technically FIDS are used for sign-in